### PR TITLE
Playwright Migration: Port Final Batch of Cypress Tests

### PR DESCRIPTION
**Pull request by ardelato**
## Description

This is the last batch of Cypress tests we need to port over to Playwright.

### Changes:

- Ported over `products-list` Cypress tests to Playwright
- Ported over the `redirects` Cypress test to Playwright
- Added a utility function to check if an element is visible within a Viewport
- Added a utility function to access nested object properties using `dot notation` and `lodash.get` - This mimics Cypress' [`its` function](https://docs.cypress.io/api/commands/its) 
- Skipped `product/pro_user` test

### CR Notes:

In order to port some of the Cypress tests while retaining the same test assertions,  I needed to add some utility functions as well as modify the selectors to target the desired elements.

Additionally, while smoke testing that the Playwright tests all work, I discovered that the `product/pro_user` test started to fail on this branch and on the `main` branch. I am not quite sure why it started to fail nor has it been easy to debug, therefore I will address it in another pull request and skip it for now.

### QA Notes:
Similar steps as https://github.com/iFixit/react-commerce/pull/1149

- [x] Also ensure there is no `test.only` statements in the Playwright test files.

Closes: #1113

### CODE REVIEW

**Comment by deltuh-vee** [comment]
There are no test.only statements in the Playwright test files. Still need to check the steps from https://github.com/iFixit/react-commerce/pull/1149.

**Comment by deltuh-vee** [comment]
[Adding external_block since no one on the QA team can test this. ](https://ifixit.slack.com/archives/C03B1PPTH43/p1671658794968309?thread_ts=1671650353.631349&cid=C03B1PPTH43)

**Comment by ardelato** [comment]
@deltuh-vee that slack channel seems to be private and I can't access it

**Comment by deltuh-vee** [comment]
> that slack channel seems to be private and I can't access it

Huh didn't realize qa-testers-squad was private. Anyways here's the relevant info:
image

**Comment by ardelato** [comment]
QA 👍 now that we integrated Playwright into our CI pipeline, we can validate the tests work. #1193

**Review by batbattur — APPROVED** [review]
CR 🚛 
Non-blocking suggestion on possibly moving `queryResponse` to playwright utils. I think we can do it in a follow-up PR if we want to.

File: frontend/tests/playwright/product/pro_user.spec.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { interceptLogin } from '../utils';
 import { test, expect } from '@playwright/test';
 
 test.describe('Pro user test', () => {
-   test.skip('Will give pro users a discount', async ({ page }) => {
+   test('Will give pro users a discount', async ({ page }) => {
```
Comment (batbattur) [inline frontend/tests/playwright/product/pro_user.spec.ts:5]: Nice! 🎉

File: frontend/tests/playwright/product-list/parts_page_search.spec.ts [hunk 2]
```diff
@@ -1,50 +1,72 @@
...
+      expect(page.getByTestId('collections-search-box')).not.toBeDisabled();
+
+      // Start listening for algolia search request
+      const queryResponse = page.waitForResponse((response) => {
+         return (
+            response.url().includes('algolia') &&
+            response.url().includes('queries?') &&
+            response.status() === 200
+         );
+      });
```
[the remark this answers is not in the document]
Comment (ardelato) [reply frontend/tests/playwright/product-list/parts_page_search.spec.ts:23]: Hmm, yeah we probably should. I will open up a new pull for this.

[the remark this answers is not in the document]
Comment (ardelato) [reply frontend/tests/playwright/product-list/parts_page_search.spec.ts:23]: https://github.com/iFixit/react-commerce/pull/1194

**Pull request merged**